### DISCUSSION TITLE
last Loop value is available in the whole component

### DISCUSSION THREAD

**Discussion by rohit-sukanfood** [livewire/livewire#5134] [1. Help] [answered]
Hi!
I have loop in my component 
`<select wire:model="unit" id="unit" class="form-select @error("unit") is-invalid @enderror">
                                <option
                                    value="">{{__("app.generals.select-", ["title" => trans_choice("app.generals.unit", 1)])}}</option>
                                @foreach($units as $u => $ut)
                                    <option value="{{$ut->id}}">{{$ut->name}}</option>
                                @endforeach
                            </select>`

but why was i able to access the last loop value just doing this `{{$ut}}`?

Note: $units is coming from database/model

**Comment by Martleby** [comment, marked as answer]
It's PHP's scoping at work. You might think that the foreach loop would introduce its own scope... but not in PHP! For example:

```<?php

$things = [
            ['name' => 'martin', 'taste' => 'good'],
            ['name' => 'paul', 'taste' => 'poor']
];

foreach($things as $key => $value){
    echo $value['name'];
}

//Will dump the last value assigned to $value
echo var_dump($value);```

**Reply by joshhanley** [reply]
@rohit-sukanfood yeah @Martleby is correct. This is more a of a PHP question than a Livewire one. Have a look at the example and even test it in an app to see how it works 🙂

**Comment by rohit-sukanfood** [comment]
I did not know that.  Thanks  @Martleby  @joshhanley  for explaining things.

**Reply by joshhanley** [reply]
@rohit-sukanfood no worries! 🙂
